EmsgBox.go

```diff
@@ -0,0 +1,126 @@
+package mp4box
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+)
+
+//FullBox - Base box holding the bytes
+/*
+aligned(8) class FullBox(unsigned int(32) boxtype, unsigned int(8) v, bit(24) f) extends Box(boxtype) {
+	unsigned int(8) version = v;
+	bit(24) flags = f;
+}
+*/
+type EmsgBox struct {
+	BaseBox
+	SchemeIdUri      string
+	SchemeIdVal      string
+	MsgData          string
+	TimeScale        []byte
+	PresentTimeDelta []byte
+	EvtDur           []byte
+	IdVal            []byte
+}
+
+//Interface methods Impl - Begin
+//getLeafBox() returns leaf object Box interface
+func (b *EmsgBox) getLeafBox() Box {
+	return b
+}
+
+//GetEmsgBox - Implement Box method for this object
+func (b *EmsgBox) GetEmsgBox() (*EmsgBox, error) {
+	b.SchemeIdUri, b.SchemeIdVal, b.MsgData, b.TimeScale, b.PresentTimeDelta, b.EvtDur, b.IdVal = b.GetAllData()
+	return b, nil
+}
+
+//Interface methods Impl - End
+
+//GetPayload - Returns the payload excluding headers
+func (b *EmsgBox) GetPayload() []byte {
+	ret := b.BaseBox.getPayload()
+	return ret
+}
+
+//GetMsgData - Returns the message data from payload
+func (b *EmsgBox) GetMsgData() string {
+	return b.MsgData
+}
+
+//GetTimeScale - Returns the timescale data from payload
+func (b *EmsgBox) GetTimeScale() []byte {
+	return b.TimeScale
+}
+
+func (b *EmsgBox) GetAllData() (schemeIdUri string, schemeIdVal string, msgData string,
+	timeScale []byte, presentTimeDelta []byte, evtDur []byte, idVal []byte) {
+	payload := b.BaseBox.getPayload()
```

Comment:
you should get FullBox Payload that strips the header and gives you the base slice with only this payload

Reply:
updated basebox payload to fullbox payload.